core/src/test/java/com/netflix/iceberg/LocalTableOperations.java

```diff
@@ -52,25 +53,19 @@ public void commit(TableMetadata base, TableMetadata metadata) {
   }
 
   @Override
-  public InputFile newInputFile(String path) {
-    return localInput(path);
-  }
-
-  @Override
-  public OutputFile newMetadataFile(String filename) {
-    try {
-      File metadataFile = temp.newFile(filename);
-      metadataFile.delete();
-      metadataFile.deleteOnExit();
-      return Files.localOutput(metadataFile);
-    } catch (IOException e) {
-      throw new RuntimeIOException(e);
-    }
+  public FileIO io() {
+    return io;
   }
 
   @Override
-  public void deleteFile(String path) {
-    new File(path).delete();
+  public String metadataFileLocation(String fileName) {
+    return createdMetadataFilePaths.computeIfAbsent(fileName, name -> {
+      try {
+        return temp.newFile(name).getAbsolutePath();
```

Comment:
Shouldn't this still run delete() and deleteOnExit()?

Reply:
You're right and that should take care of deleteOnExit. I thought that the delete was needed because it creates the file, but I guess not since tests are passing.